src/types/timeseries.h

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#pragma once
+
+#include <nonstd/span.hpp>
+#include <optional>
+
+#include "storage/redis_metadata.h"
+
+class TSChunk;
+class UncompTSChunk;
+
+using TSChunkPtr = std::shared_ptr<TSChunk>;
```

Comment:
Unless we face some dynamic lifetime issue, we should prefer `std::unique_ptr` than `std::shared_ptr`.